swap_meet/vendor.py

```diff
@@ -1,2 +1,219 @@
+        
+    #     # check if items list is empty
+    #     if len(items) == 0:
+    #         # if items list is empty return None
+    #         return None
+        
+    #     # create best_item variable to represent a single item with the best condition
+    #     # use lambda to get assert that item is the variable to iterate with and condition is the method instance to iterate through
+    #     # use max() function to get the best_item result with the highest condition
+    #     # max () by default will only return one item
+    #     best_item = max(items, key=lambda item: item.condition)
+        
+    #     return best_item
+    
+        
+    def swap_best_by_category(self, other_vendor, my_priority, their_priority):
+        
+        # create item_they_want variable to represent the item that other_vendor wants
+        # use get_best_by_category (with their_priority passed in as arugment) as an instance attirubute of self (vendor)
+        item_they_want = self.get_best_by_category(their_priority)
+        
+        # create item_i_want variable to represnt the item that self (vendor) wants 
+        # use get_best_by_category (with my_priority passed in as arugment) as an instance attirubute of other_vendor
+        item_i_want = other_vendor.get_best_by_category(my_priority) 
+        
+        # create swapped_result variable to represent the item_they_want and the item_i_want
+        # use swap_items instance method on self
+        # pass arguments other_vendor, item_they_want, item_i_want into swap_items to pass in the correct items to swap and vendor to swap with
+        swapped_result = self.swap_items(other_vendor, item_they_want, item_i_want)
+    
+        # swapped result will only take item_they_want and item_i_want as a single item
+        # because swap_items uses the max () which by default returns a single item as the item with the max value
+        return swapped_result
+
+
+
+
+    # W6- SJ - Pseudo
+    # swap the best item of certain categories with another Vendor
+    # three arguments: other_vendor, my_priority, their_priority
+    # other_vendor, which represents another Vendor instance to trade with
+    # my_priority, which represents a category that the Vendor wants to receive
+    # their_priority, which represents a category that other_vendor wants to receive
+    
+    # def  swap_best_by_category(self):
+        
+    #     # the best item in my_inventory that MATCHES their_priority category
+    #     # is swapped with 
+    #     # the best item in other_vendor's inventory that matches my_priority
+    #     # then returns True 
+        
+    #     # else if the vendor has no item that matches their_priority category, 
+    #     #  no swap happens, 
+    #     #  return False
+        
+    #     # else if the other_vendor has no item that matches my priority category,
+    #     #  no swap happens
+    #     # return False
+        
+    #     pass
```

Comment:
Great work, you all! If you have any questions about the comments I left for you, please feel free to schedule a 1:1 session with me and I would be happy to clarify! Think about what optional enhancements you could possibly make or how refactor the code effectively.  Keep up the good work! ⭐️